sdks/python/apache_beam/typehints/decorators.py

```diff
@@ -378,6 +378,43 @@ def has_simple_output_type(self):
         self.output_types and len(self.output_types[0]) == 1 and
         not self.output_types[1])
 
+  def strip_pcoll_input(self):
+    # type: () -> IOTypeHints
+
+    input_type = self.input_types[0][0]
+    if isinstance(input_type, typehints.AnyTypeConstraint):
```

Comment:
Also verify that input_type is a PCollection or PBegin.
PCollection or PDone for output type